gridt/src/app/core/settings.service.ts

```diff
@@ -22,28 +24,102 @@ export class SettingsService {
+  * - If the server is unavailable but local ID is available, emit local ID.
+  * - Only when updateIdentity() function is called should the userIdentity$
+  *   change.
+  *
+  * - Makes sure all errors are caught such that this function always gives an
+  *   output.
+  */
+  private mapErrorToEmptyID = pipe(
+    catchError( err => {console.warn(err); return of<Identity>(this.empty_identity);}),
+    flatMap( (id) => of(<Identity>(id)))
+  );
+
+  private mapErrorToFalse = pipe(
+    catchError( err => {console.warn(err); return of<boolean>(false);})
+  );
+
+  public updateIdentity(): void {
+    // Combine server and local id. Catch errors to make sure pipe runs.
+    forkJoin({
+      server: this.api.userIdentity$().pipe( this.mapErrorToEmptyID ),
+      local: this.localIdentity$.pipe( this.mapErrorToEmptyID )
+    }).pipe(
+      // Update local id when server id is different.
+      tap( (ids) => console.debug(`recieved server id = ${JSON.stringify(ids.server)} \n received local id = ${JSON.stringify(ids.local)}`)),
+      flatMap( (ids) => iif(
+        () => (
+          (JSON.stringify(ids.server)===JSON.stringify(ids.local))
+          || (JSON.stringify(ids.server) == JSON.stringify(this.empty_identity))
+        ),
+        of(true),
+        this.setLocalIdentity$(ids.server)
+      )),
+      // Make sure any errors arising from setting local identity are
+      // transformed into a warning and continue.
+      this.mapErrorToFalse,
+      // Emit local identity
+      flatMap( () => this.localIdentity$ ),
+      // If local identity unavailable emit empty ID.
+      this.mapErrorToEmptyID,
+      // make sure subscription completes and gives only one update.
+      take(1)
+    ).subscribe( (id) => this._user_identity$.next(id));
+  };
+
   /**
-   * Observable to obtain settings from secure storage
+   * Observable to obtain identity from secure storage
    * If secure storage is empty, this returns an unavailable error
    */
   public localIdentity$: Observable<Identity> = this.auth.readyAuthentication$.pipe(
     catchError( () => throwError(this.error_codes.GETIDFAIL + ": " + this.error_codes.NOTLOGGEDIN)),
     flatMap(() => this.secStore.get$("identity").pipe(
       catchError( () => throwError(this.error_codes.GETIDFAIL + ": " + this.error_codes.SECSTOREUNAVAILABLE))
-    ))
+    )),
+    flatMap( (id) => of<Identity>(id))
   );
 
   /**
-   * Store settings in the secure storage.
-   * @param identity Settings to be stored.
+   * Store identity in the secure storage.
+   * @param identity Identity to be stored.
    */
   public setLocalIdentity$(identity: Identity): Observable<boolean> {
     return this.auth.readyAuthentication$.pipe(
       catchError( () => throwError(this.error_codes.SETIDFAIL + ": " + this.error_codes.NOTLOGGEDIN)),
       take(1), // Makes sure the observable completes
+      tap( () => console.debug('Storing Identity in Localstorage')),
```

Comment:
don't forget the space at the end there aswell!
either (statement) or ( statement ) and not the ( statement)
I have made that mistake too.